implant/sliver/handlers/rpc-handlers_linux.go

```diff
@@ -63,3 +73,153 @@ func psHandler(data []byte, resp RPCResponse) {
 	data, err = proto.Marshal(psList)
 	resp(data, err)
 }
+
+func getFdFromPath(path string) (int, error) {
+	fd, err := unix.Open(path, unix.O_RDONLY|unix.O_CLOEXEC, 0)
+	if err != nil {
+		return -1, err
+	}
+	return fd, nil
+}
+
+func getUniqueFd(fd int) string {
+	// Returns the unique namespace ID
+	var s unix.Stat_t
+
+	err := unix.Fstat(fd, &s)
+
+	if err != nil {
+		return "Unknown"
+	}
+
+	return fmt.Sprintf("NS(%d:%d)", s.Dev, s.Ino)
+}
+
+func ifconfigLinuxHandler(_ []byte, resp RPCResponse) {
+	interfaces := ifconfigLinux()
+	// {{if .Config.Debug}}
+	log.Printf("network interfaces: %#v", interfaces)
+	// {{end}}
+	data, err := proto.Marshal(interfaces)
+	resp(data, err)
+}
+
+func nsLinuxIfconfig(interfaces *sliverpb.Ifconfig) {
+	namespacesFound := make(map[uint64]string)
+
+	procDir := "/proc"
+	procContents, err := ioutil.ReadDir(procDir)
+
+	if err != nil {
+		//{{if .Config.Debug}}
+		log.Printf("error reading /proc: %v", err)
+		//{{end}}
+		return
+	}
+
+	for _, entry := range procContents {
+		if !entry.IsDir() {
+			continue
+		}
+		match, _ := filepath.Match("[1-9]*", entry.Name())
+		if match {
+			// Check if /proc/PID/net/ns exists
+			checkPath := filepath.Join(procDir, entry.Name(), "/ns/net")
+
+			if _, err := os.Stat(checkPath); !os.IsNotExist(err) {
+				// path for /proc/PID/ns/net exists
+				// inode used to track unique namespaces
+				var inode uint64
+
+				fileinfo, err := os.Stat(checkPath)
+
+				if err != nil {
+					//{{if .Config.Debug}}
+					log.Printf("error : %v", err)
+					//{{end}}
+					continue
+				}
+				inode = fileinfo.Sys().(*syscall.Stat_t).Ino
+				// Track unique namespaces
+				namespacesFound[inode] = checkPath
+			}
+
+		}
+	}
+
+	// Lock the OS Thread so we don't accidentally switch namespaces
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	// Save the current network namespace
+	origns, _ := getFdFromPath(fmt.Sprintf("/proc/%d/task/%d/ns/net", os.Getpid(), unix.Gettid()))
```

Comment:
It'd be better to use `filepath.Join(...)` instead of `fmt.Sprintf` here.

Reply:
I shall fix! :)

Reply:
I fixed that and also made the output a bit more useful:

```
sliver (FAVOURABLE_FISHERMAN) > ifconfig                                                                  
                                                                                                          
+--------------------------------------+                                                                  
| eth0                                 |                                                                  
+--------------------------------------+                                                                  
| # | IP Addresses | MAC Address       |                                                                  
+---+--------------+-------------------+                                                                  
| 2 | 10.0.2.15/24 | 08:00:27:e2:ac:47 |                                                                  
+--------------------------------------+                                                                  
                                                                                                          
+---------------------------------------+                                                                 
| tap0 NS(/proc/34888/ns/net)           |                                                                 
+---------------------------------------+                                                                 
| # | IP Addresses  | MAC Address       |                                                                 
+---+---------------+-------------------+                                                                 
| 2 | 10.0.2.100/24 | 86:aa:ec:2a:af:81 |                                                                 
+---------------------------------------+                                                                 
4 adapters not shown.                                                                                     
```

The path, with PID, will give enough info to run commands like `nsenter` to enter the namespace